### DISCUSSION TITLE
Some bugs and ideas of functionality for library files and playlists

### DISCUSSION THREAD

**Discussion by raphial** [martpie/museeks#617] [Ideas]
Hello! I have some ideas for helping improve Museeks. I absolutely, 100% love and support this music player, and have really enjoyed my time with it. I appreciate all of the hard work put into it.

There's a few things I want to suggest, and noticed, about the program I wanted to discuss. For starters, I use Linux Mint, a Debian-based distro using Ubuntu PPAs with Cinnamon's desktop environment. I'm using Linux Mint 20.2.

For starters, the **Museeks** menu for me is unresponsive. I can hover it, view the menu, but none of the menu objects are clickable. I feel that's a pretty big bug. 

Now onto some suggestions that I think really, really would help the functionality of the program. Some basic function suggestions. 

**Playlists**

One fault of like 90% of the music players on Linux is that you cannot sort playlist songs by artist, album, or anything else. They're statically placed and un-sortable. I understand how playlist files and .m3u files work, but I feel like the program should be able to store a dynamically changing internal playlist file while using the original, uploaded .m3u file (For example) as a constant reference. Any changes to the original .m3u/playlist file would update the internal program playlist file. So any appearance changes, such as sorting by artist, is program-side, whereas the properties and contents of the playlist continue to stay, and be reference by, the original .m3u/playlist file. If we could create a sorting method for playlists without physically changing the original playlist file, I think this would give Museeks a HUGE advantage over all other music players available on Linux. 

As a former iTunes player fan for years and years, I really think this feature would be really appreciated by many, many people. This would really streamline Museeks for a huge audience.

**Library Files**

Another big, but basic, functionality I think would really, really help streamline Museeks is having a file-properties manager or dialog for your songs. Such as, for example, right click a song > Song Properties > change song information, artist, name, etc. There are a handful of tag-editing applications out there, but having a simple one used within Museeks would likely help a ton. 

A suggestion is you could use `mid3v2` as the backend for such a function as it uses commandline. You could create a simple Song Properties dialog and any information could be sent through mid3v2 to save to the file, streamlining the process. mid3v2 is a very simple command line tool available for most distros and their github is located [here](https://github.com/dhamaniasad/mutagen/blob/master/tools/mid3v2).

Being able to edit the song properties within the program, rather than resorting to opening another program, going through the files, etc etc would really help also streamline Museeks.

**Conclusion**

Museeks is already, in my opinion, one of the best and cleanest music programs available on the market, and has SO much potential to continue to grow. Especially for Linux. I would vouch and say its the iTunes equivalent for Linux, and there's not much of another option out there like Museeks and how Museeks interface is so simple, elegant, and streamlined. 

I haven't programmed anything before for desktop software, but I have a huge amount of knowledge for software syntax, and web languages both server side and client side. If there's anything I can do to help, let me know. I'd be willing to help invest my time into your project if you need additional man-power.

I also highly encourage having a Discord server, or something similar, to actively interact with clients using the program, as it's gotten a lot more love and has a huge audience. This could also help streamline Museeks' support and reputation.

Thanks for investing so much time into creating an excellent music player regardless!

**Comment by martpie** [comment]
> its the iTunes equivalent for Linux

That's a funny thing to say, because it was absolutely my goal at the beginning 😄 Have a pretty music player on Linux (maybe not with all iTunes features though).

> Library Files

cf. #619, it's coming (probably).

> Playlists

So this is an interesting idea, I haven't thought about that. The main issue I see is how would people "rollback" to the "no-custom sorting" state. An additional button on the UI or something, but that may add some clutter, so I want to be a bit cautious there (basically, I usually prefer to not implement a feature, than implement a feature that don't meet my UI/UX standards).
